ciri/main.c

```diff
@@ -9,13 +9,15 @@
 
 #include "ciri/core.h"
 #include "ciri/node.h"
+#include "ciri/conf/provider.h"
 #include "utils/logger_helper.h"
 
 #define MAIN_LOGGER_ID "main"
 
 static core_t core_g;
+static ciri_config_t* config;
```

Comment:
Just store it in the core so that we don't have to pass it everywhere

Reply:
Agree, but we can't get `config` if we wrapped it in core in `node.c`, since we forward declared core here.

We'll need it in `bool receiver_init(receiver_state_t *const state, node_t *const node, uint16_t tcp_port, uint16_t udp_port);`

Reply:
I'm not sure to understand what you mean. Node has access to the core, then node will have access to config.